frontend/src/metabase/visualizations/echarts/cartesian/waterfall/model/dataset.ts

```diff
@@ -7,103 +7,69 @@ import {
 import { checkNumber } from "metabase/lib/types";
 
 import type { ComputedVisualizationSettings } from "metabase/visualizations/types";
-import type {
-  WaterfallDataset,
-  WaterfallDatum,
-  WaterfallEmptyValue,
-} from "../types";
-import { DATASET_DIMENSIONS } from "../constants";
-
-function createDatum({
-  dimension,
-  barOffset,
-  increase,
-  decrease,
-  total,
-}: {
-  dimension: string;
-  barOffset?: number;
-  increase?: number | WaterfallEmptyValue;
-  decrease?: number | WaterfallEmptyValue;
-  total?: number;
-}): WaterfallDatum {
-  return {
-    [DATASET_DIMENSIONS.dimension]: dimension,
-    [DATASET_DIMENSIONS.barOffset]: barOffset ?? 0,
-    [DATASET_DIMENSIONS.increase]: increase ?? "-",
-    [DATASET_DIMENSIONS.decrease]: decrease ?? "-",
-    [DATASET_DIMENSIONS.total]: total ?? "-",
-  };
-}
+import type { WaterfallDataset } from "../types";
 
 export function getWaterfallDataset(
   rows: RowValues[],
   cardColumns: CartesianChartColumns,
-  negativeTranslation: number,
-  total: number,
   settings: ComputedVisualizationSettings,
-): WaterfallDataset {
+) {
   const columns = assertMultiMetricColumns(cardColumns);
   const dataset: WaterfallDataset = [];
 
+  // Step 1: calculate runningSums, negativeTranslation beforehand
```

Comment:
I changed the approach to draw the bars based on the cumulative sum at each bar, because the pow scale applies the square root function to the sum, not to the individual increase/decrease bar.